src/Fantomas.Core/SourceParser.fs

```diff
@@ -608,10 +608,26 @@ let (|Sequential|_|) =
     | SynExpr.Sequential (_, isSeq, e1, e2, _) -> Some(e1, e2, isSeq)
     | _ -> None
 
-let rec (|Sequentials|_|) =
+let (|Sequentials|_|) =
```

Comment:
We have solved this type of problem before in the codebase, this is quite the departure from that.

To make it tail recursive you can use the following:

```fsharp
let (|Sequentials|_|) e =
    let rec visit (e: SynExpr) (finalContinuation: SynExpr list -> SynExpr list) : SynExpr list =
        match e with
        | Sequential (e1, e2, _) -> visit e2 (fun xs -> e1 :: xs |> finalContinuation)
        | e -> finalContinuation [ e ]

    match e with
    | Sequential (e1, e2, _) ->
        let xs = visit e2 id
        Some(e1 :: xs)
    | _ -> None
```